src/main/java/org/folio/rest/impl/GetPOLineByIdHelper.java

```diff
@@ -0,0 +1,114 @@
+package org.folio.rest.impl;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import me.escoffier.vertx.completablefuture.VertxCompletableFuture;
+import org.folio.orders.rest.exceptions.HttpException;
+import org.folio.orders.utils.HelperUtils;
+import org.folio.rest.jaxrs.model.Error;
+import org.folio.rest.jaxrs.model.Errors;
+import org.folio.rest.jaxrs.model.PoLine;
+import org.folio.rest.jaxrs.resource.Orders;
+import org.folio.rest.tools.client.interfaces.HttpClientInterface;
+
+import javax.ws.rs.core.Response;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+
+public class GetPOLineByIdHelper {
+
+  private static final Logger logger = LoggerFactory.getLogger(GetOrdersByIdHelper.class);
+
+  public static final String MOCK_DATA_PATH = "mockdata/lines/c0d08448-347b-418a-8c2f-5fb50248d67e.json";
+  public static final String ORDER_REFERENCE_ERROR_MESSAGE = "Current order do not referenced to requested PO line";
+  public static final String ERROR_CODE_422 = "422";
+
+  private final HttpClientInterface httpClient;
+  private final Context ctx;
+  private final Handler<AsyncResult<Response>> asyncResultHandler;
+  private final Map<String, String> okapiHeaders;
+
+  public GetPOLineByIdHelper(HttpClientInterface httpClient, Map<String, String> okapiHeaders,
+                             Handler<AsyncResult<Response>> asyncResultHandler, Context ctx) {
+    this.httpClient = httpClient;
+    this.okapiHeaders = okapiHeaders;
+    this.ctx = ctx;
+    this.asyncResultHandler = asyncResultHandler;
+  }
+
+  public CompletableFuture<PoLine> getCompositePOLineByPOLineId(String orderId, String polineId, String lang) {
+    CompletableFuture<PoLine> future = new VertxCompletableFuture<>(ctx);
+
+    HelperUtils.getCompositePoLineById(polineId, lang, httpClient, ctx, okapiHeaders, logger)
```

Comment:
Instead of calling `getCompositePoLineById` probably it makes sense to call ` getPoLineById` and validate order id match. So in this case no extra calls will be done to storage for such case.

Reply:
Fixed. Thanks for feedback